app/controllers/admin/edition_images_controller.rb

```diff
@@ -22,6 +22,20 @@ def update
     end
   end
 
+  def create
+    new_image = @edition.images.build
+    new_image.build_image_data(image_params["image_data"])
+
+    if @edition.save && new_image.valid?
```

Comment:
In fact, I'm wondering if this would actually be better as:

```suggestion
    @new_image = @edition.images.build
    @new_image.build_image_data(image_params["image_data"])

    if @new_image.save
```

Since `@new_image` is already associated with the `@edition` – because it was created with `@edition.images.build`.

And then in the view, you'd be able to use:

```ruby
errors_for(@new_image.errors, :"image_data.file")
```